### PR TITLE
Add API documentation

### PR DESCRIPTION
**Pull request by thomas-mc-work**
This is a draft to describe the API. I think this will be very valuable for developers of clients.

regarding #92

### CODE REVIEW

**Comment by jancborchardt** [comment]
Also here: @aszlig @Galbar since @tacruc @eneiluj are a bit busy atm, and you opened pull requests for Maps too (yay! :) it would be lovely if you could review this too. Thanks! :guitar:

File: openapi.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          description: Success
+    post:
+      operationId: add-favorites
+      summary: 'Add a favorite'
+      responses:
+        '200':
+          description: Success
+      requestBody:
+        $ref: '#/components/requestBodies/favoriteBody'
+  /favorites/{id}:
```
Comment (aszlig) [inline openapi.yml]: Missing a [`parameters` object](https://swagger.io/specification/#parameterObject) here for `put` and `delete` with `in: path`.

Comment (thomas-mc-work) [reply openapi.yml]: Done.

**Comment by thomas-mc-work** [comment]
How can I add the _DCO_ now?

**Comment by jancborchardt** [comment]
To define the favorites API we should really make sure to loop in the people of other Maps apps to coordinate:

cc @vshcherb @Chumva @nazar-kutz @max-klaus of OsmAnd, and ref their API https://github.com/osmandapp/osmand-api-demo#command-addfavorite
cc @Sumaid @mlundblad from GNOME, and ref **Provide APIs to sync with gnome-maps #30** where we discussed already – @thomas-mc-work did you check that out? :)

Who else?

**Comment by thomas-mc-work** [comment]
> define the favorites API

This PR is not about definition at all. It's only about documenting what is already existing. You are free to change or discuss the API at any time after merging.

**Pull request merged**